src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -257,4 +273,86 @@ private static (SyntaxNode? Start, SyntaxNode? End) FindContainingSiblingPair(Sy
 
         return null;
     }
+
+    private static void AddComponentDependenciesInRange(SyntaxNode root, int extractStart, int extractEnd, ExtractToComponentCodeActionParams actionParams)
+    {
+        var components = new HashSet<string>();
+        var extractSpan = new TextSpan(extractStart, extractEnd - extractStart);
+
+        foreach (var node in root.DescendantNodes())
+        {
+            if (IsMarkupTagHelperElement(node, extractSpan))
+            {
+                var tagHelperInfo = GetTagHelperInfo(node);
+                if (tagHelperInfo is not null)
+                {
+                    AddDependenciesFromTagHelperInfo(tagHelperInfo, components, actionParams);
+                }
+            }
+        }
+    }
+
+    private static bool IsMarkupTagHelperElement(SyntaxNode node, TextSpan extractSpan)
+    {
+        return node is MarkupTagHelperElementSyntax markupElement &&
+                extractSpan.Contains(markupElement.Span);
+    }
+
+    private static TagHelperInfo? GetTagHelperInfo(SyntaxNode node)
+    {
+        if (node is MarkupTagHelperElementSyntax markupElement)
+        {
+            return markupElement.TagHelperInfo;
+        }
+
+        return null;
+    }
+
+    private static void AddDependenciesFromTagHelperInfo(TagHelperInfo tagHelperInfo, HashSet<string> components, ExtractToComponentCodeActionParams actionParams)
+    {
+        foreach (var descriptor in tagHelperInfo.BindingResult.Descriptors)
+        {
+            if (descriptor is not null)
+            {
+                foreach (var metadata in descriptor.Metadata)
+                {
+                    if (metadata.Key == TagHelperMetadata.Common.TypeNamespace &&
+                        metadata.Value is not null &&
+                        !components.Contains(metadata.Value))
+                    {
+                        components.Add(metadata.Value);
+                        actionParams.Dependencies.Add($"@using {metadata.Value}");
```

Comment:
I don't think `Dependencies` is the right name for this, though I think perhaps its been removed in a follow up anyway. I would just call it `UsingDirectives` though. I'd also consider whether its simpler for the Params type to just use a string, but then again if it's been removed in a future PR this whole comment is mute.

Reply:
This has been renamed to `ComponentDependencies`

Reply:
Honesty, that's just leaning in to being incorrect. They were never dependencies, and they certainly aren't components. Objectively this is a list of using directives. Unless those using directives are in the witness protection program, we should call them what they are.